cloudformation.yaml

```diff
@@ -358,84 +353,103 @@ Resources:
           - |
            {
              "Comment": "Elasticsearch Node Rotation",
-             "StartAt": "AutoScalingGroupCheck",
+             "StartAt": "DiscoverAutoScalingGroup",
              "States": {
-               "AutoScalingGroupCheck": {
-                 "Type": "Task",
-                 "Resource": "${AutoScalingGroupCheckArn}",
-                 "Next": "GetOldestNode"
+                "DiscoverAutoScalingGroup": {
+                  "Type": "Task",
+                  "Resource": "${DiscoverAutoScalingGroupArn}",
+                  "Next": "CheckNotAlreadyRunning"
+                },
+                "CheckNotAlreadyRunning": {
+                  "Type": "Choice",
+                  "Choices": [
+                    {
+                      "Variable": "$.alreadyRunning",
+                      "BooleanEquals": false,
+                      "Next": "AutoScalingGroupCheck"
+                    }
+                  ],
+                  "Default": "StopAsAlreadyRunning"
```

Comment:
Yeah sorry I will expand on this a bit more in the PR description but the idea is set to set a more aggressive schedule rather than have separate schedules for each autoscaling group. So it'll become more likely to run in parallel so we just shrug it off and wait for the next trigger

Reply:
One potential problem here is that a single long-running execution in the account will now silently block all other node rotations. We could consider using [`TimeoutSeconds`](https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-state-machine-structure.html) to mitigate this, as the AWS default is [_very_ patient](https://docs.aws.amazon.com/step-functions/latest/dg/limits.html#service-limits-state-machine-executions) (I'd love to know the use-case for a year-long execution)! I think we'd expect any rotation to complete within ~24 hours?

Reply:
Having looked at this in more detail, I can see that all of our lambdas have a sensible timeout and all of the retry states seem to have `MaxAttempts` defined, so I think it's safe to ignore my comment above.

Reply:
Ah cool, no worries. It's a definitely a balance, in an ideal world I guess we'd discover all "clusters" in an account and rotate them independently. At the moment though our best proxy for that is `Stack`, `App` and `Stage` tags and it felt simpler to be unplugged from them entirely. I'd be happy to switch strategies if we hit issues.

> I'd love to know the use-case for a year-long execution

Yeah it's pretty wild. I think it's to support their [callback pattern](https://aws.amazon.com/about-aws/whats-new/2019/05/aws-step-functions-support-callback-patterns/) where you can proceed if a token is returned. Things like "delete this user unless they respond to an email within blah amount of time"